### DISCUSSION TITLE
How filter parameters for Report Portal 5.5.0 works?

### DISCUSSION THREAD

**Discussion by Itso-Dimitrov-Syncplicity** [reportportal/reportportal#1512] [Q&A] [answered]
In Build: 4.3.0 we can specify directly in the URL parameters to create filters for specific report portal record
**Ex**: launches/New_filter?page.page=1&page.size=50&filter.has.tags=Jenkins_Run:{BuildNumber}

for Build: 5.5.0 i tried with  - launches?filter.has.attributeValue={BuildNumber}&filter.has.attributeKey=Jenkins_Run'
but that did not work. 

How we can specify such for Build: 5.5.0?

**Comment by evjlobanova** [comment, marked as answer]
In version 5.5 there is no such functionality.  
Right now, I can propose a workaround. 
You can create a basic filter for instance based on the launch name. 
Then you can create a widget Component Health Check View
![image](https://user-images.githubusercontent.com/30413511/134316398-e45a74a2-adb9-4256-a3ab-857e83d4b6d4.png)
And specify for this widgets option: key value Build
![image](https://user-images.githubusercontent.com/30413511/134316574-3e62e359-80c2-447b-bfc3-31ed9ab2e246.png)

In this case the system will group launches and tests around attributes with key *Build* and with unique Values. 
This widget has clickable links, so that if you click on the panel 
![image](https://user-images.githubusercontent.com/30413511/134316880-c5db23c4-7446-4adf-ba3a-93def60c471b.png)
The system redirects you to the page with tests from launches with specified build:
![image](https://user-images.githubusercontent.com/30413511/134317064-a22a654f-7fa7-440f-85fc-9fc258d385ae.png)

And now the workaround)):
The link has the format:

https://localhost:8080/ui/#Project_Name/launches/6264/list?item0Params=filter.eq.hasStats%3Dtrue%26filter.eq.hasChildren%3Dfalse%26filter.in.type%3DSTEP%26filter.has.compositeAttribute%3Dbuild%253A3.23.11.37.18%26launchesLimit%3D600%26isLatest%3Dfalse%26filter.in.status%3DPASSED%252CFAILED%252CSKIPPED%252CINTERRUPTED

So that if you have already created a widget with grouping around the build, you will be able to generate a link to the new builds.

In version 5.6 - we will provide a more convenient way to do this operation.

**Reply by Itso-Dimitrov-Syncplicity** [reply]
Thank you for the reply @evjlobanova :)
I tried your suggestion and it worked, but is there a way to be redirected on the page launches not in the test results with specified build number in the url? 
Ex:
![image](https://user-images.githubusercontent.com/53829037/162737102-d4938eb2-9b05-4901-b06d-3a7258ad3d07.png)

**Comment by mcookson3** [comment]
@evjlobanova You mentioned above that there would be a more convenient way to do this in 5.6, but I looked at the release notes for 5,6 and I don't see anything about a fix for this. Is that still in the works?

**Reply by pulse00** [reply]
@evjlobanova same here - any info on this? 5.7 does also not mention this issue.

**Comment by Zenin-Dmitry** [comment]
@patsevanton I also find the launch filter by attribute in the url useful. I run a lot of tests where one of the attributes is $CI_COMMIT_SHORT_SHA.
I want to generate a link to the passage of all test runs with this attribute. This link will then be included in the email template for testers.

**Comment by JaneIRL** [comment]
For anyone who gets here in the future: https://github.com/reportportal/reportportal/issues/2339#issuecomment-2443482629
